### ISSUE TITLE
Bug: empty arrays error in `var`, `sdev` and `evalcorr`

### ISSUE THREAD

**Issue by Gattocrucco** [gplepage/gvar#24]
```python
import gvar
gvar.mean([])
gvar.evalcov([])
gvar.evalcov_blocks([])
# all ok
```

```python
print(repr(gvar.var([])))
```
```
array(None, dtype=object)
```

which causes

```python
gvar.sdev([])
```
```
File _utilities.pyx:212, in gvar._utilities.sdev()

TypeError: unsupported operand type(s) for ** or pow(): 'NoneType' and 'float'
```

```python
gvar.evalcorr([])
```
```
File _utilities.pyx:456, in gvar._utilities.evalcorr()

IndexError: index 0 is out of bounds for axis 1 with size 0
```

It is a bit weird to use these functions on empty arrays, however it happened to me while emptying an array as a quick check to debug a fit.

**Comment by Gattocrucco** [comment]
Thanks.

**Comment by gplepage** [comment]
Fixed in v11.9.7.

**Comment by Gattocrucco** [comment]
Thank you.

I found another minor nitpick on this: the shape of the arrays returned by `evalcov` and `evalcorr` is `(1, 0)` instead of `(0, 0)`:

```python
import gvar
gvar.evalcov([])
```

```
array([], shape=(1, 0), dtype=float64)
```

**Comment by gplepage** [comment]
I don't see that it makes much difference. ``evalcov`` is returning ``[[]]`` which isn't a bad choice for an empty matrix.

**Comment by Gattocrucco** [comment]
As in the case where `evalcorr([])` caused an error, it does not make a difference most of the time but can break some code in corner cases. A covariance matrix is symmetric so the shape should be (0, 0). Imagine this example: the output of `evalcov` is passed to a function that does something on the covariance matrix. This function could realistically take `len(cov)` as the number of variables, so it would assume that the matrix is 1x1 in our case and then raise an exception as it tries to access `cov[0, 0]`.

**Comment by gplepage** [comment]
Sounds right.

**Comment by gplepage** [comment]
Fixed in v11.10.

**Comment by Gattocrucco** [comment]
Thanks.
